diff --git a/cmake/templates/OpenCVConfig-CUDALanguage.cmake.in b/cmake/templates/OpenCVConfig-CUDALanguage.cmake.in
index 259141006a..efc2516ffc 100644
--- a/cmake/templates/OpenCVConfig-CUDALanguage.cmake.in
+++ b/cmake/templates/OpenCVConfig-CUDALanguage.cmake.in
@@ -17,7 +17,7 @@ if(NOT CUDAToolkit_FOUND)
       set(CUDA_PATH "/usr/local/cuda" CACHE INTERNAL "")
       set(ENV{CUDA_PATH} ${CUDA_PATH})
     endif()
-    find_package(CUDAToolkit ${OpenCV_CUDA_VERSION} EXACT REQUIRED)
+    find_package(CUDAToolkit REQUIRED)
   else()
     message(FATAL_ERROR "Using OpenCV compiled with CUDA as first class language requires CMake \>= 3.18.")
   endif()
@@ -25,7 +25,4 @@ else()
   if(CUDAToolkit_FOUND)
     set(CUDA_VERSION_STRING ${CUDAToolkit_VERSION})
   endif()
-  if(NOT CUDA_VERSION_STRING VERSION_EQUAL OpenCV_CUDA_VERSION)
-      message(FATAL_ERROR "OpenCV library was compiled with CUDA ${OpenCV_CUDA_VERSION} support. Please, use the same version or rebuild OpenCV with CUDA ${CUDA_VERSION_STRING}")
-  endif()
 endif()
